h2/src/main/org/h2/table/TableFilter.java

```diff
@@ -778,7 +778,7 @@ public String getPlanSQL(boolean isJoin) {
             return buff.toString();
         }
         if (table.isView() && ((TableView) table).isRecursive()) {
-            buff.append(table.getName());
+            buff.append(table.getSchema().getSQL()).append('.').append(table.getName());
```

Comment:
It looks like `buff.append(table.getSQL());` should be used unconditionally instead. This if-else statement is useless after such change.

Reply:
Yes, have already tested and the `buff.append(table.getSQL())` do not do the same thing than `buff.append(table.getSchema().getSQL()).append('.').append(table.getName());`.

Reply:
Sorry, I didn't check that this method is overridden in `TableView` in incompatible way.

Use `.append(Parser.quoteIdentifier(table.getName()))` instead of to fix this place properly. Schema name is  already quoted in `Schema.getSQL()` and it should not be quoted again.

The whole condition is weird enough. It was introduced in PR #302. I'm not sure how to resolve it properly.

This place can also have the similar problem with missing schema name and unquoted view name:
https://github.com/h2database/h2database/blob/7a2a54468fdf843b88686ed60e011cf4680f2d68/h2/src/main/org/h2/command/dml/Select.java#L1154-L1161
It's better to fix it too.